[Bug] Running Test Files. Cannot Locate Test Data

# 🐛 Bug
When running the test_csv_logger.py file, it is unable to locate the test data folder which contains the relevant test files. 

## To reproduce

pytest tests/utils/test_csv_logger.py -v


** Stack trace/error message **
```
E           FileNotFoundError: [Errno 2] No such file or directory: 'tests/test_data/parameters.json'

kale/utils/csv_logger.py:66: FileNotFoundError```

## Expected Behaviour

I expected the csv_logger test to pass and for the script to be able to access the test data.

## Environment

PyTorch version: 1.9.0
Is debug build: False
CUDA used to build PyTorch: None
ROCM used to build PyTorch: N/A

OS: macOS 10.15.7 (x86_64)
GCC version: Could not collect
Clang version: 12.0.0 (clang-1200.0.32.2)
CMake version: Could not collect
Libc version: N/A

Python version: 3.8.8 (default, Apr 13 2021, 12:59:45)  [Clang 10.0.0 ] (64-bit runtime)
Python platform: macOS-10.15.7-x86_64-i386-64bit
Is CUDA available: False
CUDA runtime version: No CUDA
GPU models and configuration: No CUDA
Nvidia driver version: No CUDA
cuDNN version: No CUDA
HIP runtime version: N/A
MIOpen runtime version: N/A

Versions of relevant libraries:
[pip3] numpy==1.20.1
[pip3] numpydoc==1.1.0
[pip3] pytorch-lightning==1.3.8
[pip3] pytorch-memlab==0.2.3
[pip3] torch==1.9.0
[pip3] torchaudio==0.9.0a0+33b2469
[pip3] torchmetrics==0.4.1
[pip3] torchsummary==1.5.1
[pip3] torchvision==0.10.0
[conda] blas                      1.0                         mkl  
[conda] ffmpeg                    4.3                  h0a44026_0    pytorch
[conda] mkl                       2021.2.0           hecd8cb5_269  
[conda] mkl-service               2.3.0            py38h9ed2024_1  
[conda] mkl_fft                   1.3.0            py38h4a7008c_2  
[conda] mkl_random                1.2.1            py38hb2f4e1b_2  
[conda] numpy                     1.20.1           py38hd6e1bb9_0  
[conda] numpy-base                1.20.1           py38h585ceec_0  
[conda] numpydoc                  1.1.0              pyhd3eb1b0_1  
[conda] pytorch                   1.9.0                   py3.8_0    pytorch
[conda] pytorch-lightning         1.3.8                    pypi_0    pypi
[conda] pytorch-memlab            0.2.3                    pypi_0    pypi
[conda] torch                     1.9.0                    pypi_0    pypi
[conda] torchaudio                0.9.0                      py38    pytorch
[conda] torchmetrics              0.4.1                    pypi_0    pypi
[conda] torchsummary              1.5.1                    pypi_0    pypi
[conda] torchvision               0.10.0                   pypi_0    pypi

## Additional context
Add any other context about the problem here.

@haipinglu 
1. I did not post the entire output. The entire output is:

========================================================== test session starts ===========================================================
platform darwin -- Python 3.8.8, pytest-6.2.3, py-1.10.0, pluggy-0.13.1 -- /Users/kennedydike/opt/anaconda3/bin/python
cachedir: .pytest_cache
rootdir: /Users/kennedydike/Downloads/pykale-main
plugins: cov-2.12.1, anyio-2.2.0, nbval-0.9.6
collected 1 item                                                                                                                          

tests/utils/test_csv_logger.py::test_csv_logger FAILED                                                                              [100%]

================================================================ FAILURES =================================================================
_____________________________________________________________ test_csv_logger _____________________________________________________________

download_path = 'tests/test_data'

    def test_csv_logger(download_path):
        train_params = {"param1": 1, "param2": 2}
        method = "DANN"
        seed = 32  # To define in conftest later?
>       testing_logger, results, _, test_csv_file = setup_logger(train_params, download_path, method, seed)

tests/utils/test_csv_logger.py:12: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
kale/utils/csv_logger.py:91: in setup_logger
    record_hashes(hash_file, params_hash, train_params)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

hash_file = 'tests/test_data/parameters.json', hash_ = 'dddc9499c99f608d5d4809c4d84f68dd', value = {'param1': 1, 'param2': 2}

    def record_hashes(hash_file, hash_, value):
        """Record the hash and assoicated (training) parameters
    
        Args:
            hash_file (string): the (json) file for recording hash info
            hash_ (hash): the hash
            value (dictionary): the (unique parameter) setting
        """
        if os.path.exists(hash_file):
            with open(hash_file, "r") as fd:
                known_hashes = json.load(fd)
        else:
            known_hashes = {}
    
        if hash_ not in known_hashes:
            known_hashes[hash_] = value
>           with open(hash_file, "w") as fd:
E           FileNotFoundError: [Errno 2] No such file or directory: 'tests/test_data/parameters.json'

kale/utils/csv_logger.py:66: FileNotFoundError
============================================================ warnings summary =============================================================
../../opt/anaconda3/lib/python3.8/site-packages/tensorboard/compat/tensorflow_stub/dtypes.py:572
  /Users/kennedydike/opt/anaconda3/lib/python3.8/site-packages/tensorboard/compat/tensorflow_stub/dtypes.py:572: DeprecationWarning: `np.object` is a deprecated alias for the builtin `object`. To silence this warning, use `object` by itself. Doing this will not modify any behavior and is safe. 
  Deprecated in NumPy 1.20; for more details and guidance: https://numpy.org/devdocs/release/1.20.0-notes.html#deprecations
    (np.object, string),

../../opt/anaconda3/lib/python3.8/site-packages/tensorboard/compat/tensorflow_stub/dtypes.py:573
  /Users/kennedydike/opt/anaconda3/lib/python3.8/site-packages/tensorboard/compat/tensorflow_stub/dtypes.py:573: DeprecationWarning: `np.bool` is a deprecated alias for the builtin `bool`. To silence this warning, use `bool` by itself. Doing this will not modify any behavior and is safe. If you specifically wanted the numpy scalar type, use `np.bool_` here.
  Deprecated in NumPy 1.20; for more details and guidance: https://numpy.org/devdocs/release/1.20.0-notes.html#deprecations
    (np.bool, bool),

../../opt/anaconda3/lib/python3.8/site-packages/tensorboard/compat/tensorflow_stub/dtypes.py:597
  /Users/kennedydike/opt/anaconda3/lib/python3.8/site-packages/tensorboard/compat/tensorflow_stub/dtypes.py:597: DeprecationWarning: `np.object` is a deprecated alias for the builtin `object`. To silence this warning, use `object` by itself. Doing this will not modify any behavior and is safe. 
  Deprecated in NumPy 1.20; for more details and guidance: https://numpy.org/devdocs/release/1.20.0-notes.html#deprecations
    types_pb2.DT_STRING: np.object,

../../opt/anaconda3/lib/python3.8/site-packages/tensorboard/compat/tensorflow_stub/dtypes.py:601
  /Users/kennedydike/opt/anaconda3/lib/python3.8/site-packages/tensorboard/compat/tensorflow_stub/dtypes.py:601: DeprecationWarning: `np.bool` is a deprecated alias for the builtin `bool`. To silence this warning, use `bool` by itself. Doing this will not modify any behavior and is safe. If you specifically wanted the numpy scalar type, use `np.bool_` here.
  Deprecated in NumPy 1.20; for more details and guidance: https://numpy.org/devdocs/release/1.20.0-notes.html#deprecations
    types_pb2.DT_BOOL: np.bool,

../../opt/anaconda3/lib/python3.8/site-packages/tensorboard/compat/tensorflow_stub/dtypes.py:618
  /Users/kennedydike/opt/anaconda3/lib/python3.8/site-packages/tensorboard/compat/tensorflow_stub/dtypes.py:618: DeprecationWarning: `np.object` is a deprecated alias for the builtin `object`. To silence this warning, use `object` by itself. Doing this will not modify any behavior and is safe. 
  Deprecated in NumPy 1.20; for more details and guidance: https://numpy.org/devdocs/release/1.20.0-notes.html#deprecations
    types_pb2.DT_STRING_REF: np.object,

../../opt/anaconda3/lib/python3.8/site-packages/tensorboard/compat/tensorflow_stub/dtypes.py:623
  /Users/kennedydike/opt/anaconda3/lib/python3.8/site-packages/tensorboard/compat/tensorflow_stub/dtypes.py:623: DeprecationWarning: `np.bool` is a deprecated alias for the builtin `bool`. To silence this warning, use `bool` by itself. Doing this will not modify any behavior and is safe. If you specifically wanted the numpy scalar type, use `np.bool_` here.
  Deprecated in NumPy 1.20; for more details and guidance: https://numpy.org/devdocs/release/1.20.0-notes.html#deprecations
    types_pb2.DT_BOOL_REF: np.bool,

../../opt/anaconda3/lib/python3.8/site-packages/tensorboard/util/tensor_util.py:113
  /Users/kennedydike/opt/anaconda3/lib/python3.8/site-packages/tensorboard/util/tensor_util.py:113: DeprecationWarning: `np.object` is a deprecated alias for the builtin `object`. To silence this warning, use `object` by itself. Doing this will not modify any behavior and is safe. 
  Deprecated in NumPy 1.20; for more details and guidance: https://numpy.org/devdocs/release/1.20.0-notes.html#deprecations
    np.object: SlowAppendObjectArrayToTensorProto,

../../opt/anaconda3/lib/python3.8/site-packages/tensorboard/util/tensor_util.py:114
  /Users/kennedydike/opt/anaconda3/lib/python3.8/site-packages/tensorboard/util/tensor_util.py:114: DeprecationWarning: `np.bool` is a deprecated alias for the builtin `bool`. To silence this warning, use `bool` by itself. Doing this will not modify any behavior and is safe. If you specifically wanted the numpy scalar type, use `np.bool_` here.
  Deprecated in NumPy 1.20; for more details and guidance: https://numpy.org/devdocs/release/1.20.0-notes.html#deprecations
    np.bool: SlowAppendBoolArrayToTensorProto,

-- Docs: https://docs.pytest.org/en/stable/warnings.html
========================================================= short test summary info =========================================================
FAILED tests/utils/test_csv_logger.py::test_csv_logger - FileNotFoundError: [Errno 2] No such file or directory: 'tests/test_data/parame...
===================================================== 1 failed, 8 warnings in 15.76s ======================================================

2. Yes I get the same error when line 30 is commented

@bobturneruk Yes when the whole test suite is run this error does not occur. However I get this error:

================================================================= ERRORS ==================================================================
______________________________________________________ ERROR collecting test session ______________________________________________________
../../opt/anaconda3/lib/python3.8/site-packages/_pytest/config/__init__.py:570: in _importconftest
    mod = import_path(conftestpath, mode=importmode)
../../opt/anaconda3/lib/python3.8/site-packages/_pytest/pathlib.py:544: in import_path
    raise ImportPathMismatchError(module_name, module_file, path)
E   _pytest.pathlib.ImportPathMismatchError: ('tests.conftest', '/Users/kennedydike/Downloads/pykale-main/tests/conftest.py', PosixPath('/Users/kennedydike/Downloads/pykale-main/pykale/pykale/tests/conftest.py'))
========================================================= short test summary info =========================================================
ERROR  - _pytest.pathlib.ImportPathMismatchError: ('tests.conftest', '/Users/kennedydike/Downloads/pykale-main/tests/conftest.py', Posix...
!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!! Interrupted: 1 error during collection !!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!
============================================================ 1 error in 1.16s =============================================================

--- a/tests/conftest.py
+++ b/tests/conftest.py
@@ -10,6 +10,7 @@ from kale.utils.download import download_file_by_url
 @pytest.fixture(scope="session")
 def download_path():
     path = os.path.join("tests", "test_data")
+    os.makedirs(path,exist_ok=True)
     return path
 
 

--- a/tests/conftest.py
+++ b/tests/conftest.py
@@ -10,7 +10,7 @@ from kale.utils.download import download_file_by_url
 @pytest.fixture(scope="session")
 def download_path():
     path = os.path.join("tests", "test_data")
-    os.makedirs(path,exist_ok=True)
+    os.makedirs(path, exist_ok=True)
     return path
 
 

